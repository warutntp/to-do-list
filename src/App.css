.text-wrap-break {
  word-break: break-word;
  white-space: normal;
}

.bi-danger {
  color: red;
}

/* ตัวอย่างการปรับแต่ง CSS */
.modal-header {
  background-color: #f8f9fa;
  border-bottom: 1px solid #dee2e6;
}

.modal-title {
  color: #495057;
}

.modal-footer {
  display: flex;
  justify-content: space-between;
}

.modal-footer .btn {
  min-width: 100px;
}

.modal-footer .btn-danger {
  background-color: #dc3545;
  border-color: #dc3545;
}

.modal-footer .btn-secondary {
  background-color: #6c757d;
  border-color: #6c757d;
}