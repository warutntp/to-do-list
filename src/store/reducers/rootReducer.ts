import { TodoItemModels } from "../../types/TodoItemModels";

export interface RootState {
  todos: TodoItemModels.TodoItemModel[];
}
