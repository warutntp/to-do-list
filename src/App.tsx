import React from "react";
import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";
import "bootstrap-icons/font/bootstrap-icons.css";
import TodoListPage from "./pages/TodoListPage";

const App: React.FC = () => {
  return <TodoListPage />;
};

export default App;
