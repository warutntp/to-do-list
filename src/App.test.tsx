import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import App from "./App";
import { RootState } from "./store/reducers/rootReducer";

const mockStore = configureStore([]);

const initialState: RootState = {
  todos: [],
};

describe("App", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it("should render App correctly", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    expect(screen.getByText("To-do List App")).toBeInTheDocument();
    expect(screen.getByText("Add Todo")).toBeInTheDocument();
  });
});
