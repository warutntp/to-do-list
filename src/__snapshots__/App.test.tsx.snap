// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-4"
  >
    <div
      class="d-flex justify-content-center mb-4"
    >
      <h3>
        To-do List App
      </h3>
    </div>
    <div
      class="d-flex justify-content-center bg-primary mb-4 p-5 rounded"
    >
      <div
        class="todo-form-container bg-white p-3 rounded col-12 col-md-9 col-lg-4"
      >
        <form
          class="mb-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="title"
            >
              Title
            </label>
            <input
              class="form-control placeholder-italic"
              id="title"
              placeholder="Please enter todo title"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="description"
            >
              Description (Optional)
            </label>
            <textarea
              class="form-control placeholder-italic"
              id="description"
              placeholder="Enter todo description"
              rows="3"
            />
          </div>
          <div
            class="mb-3"
          >
            <button
              class="btn btn-primary w-100"
              type="submit"
            >
              Add Task
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="d-flex justify-content-end align-items-center mb-4"
    >
      <div
        class="btn-group"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          All
        </button>
        <button
          class="btn btn-outline-primary"
          type="button"
        >
          Incomplete
        </button>
        <button
          class="btn btn-outline-primary"
          type="button"
        >
          Completed
        </button>
      </div>
    </div>
    <hr />
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover table-striped"
      >
        <thead>
          <tr>
            <th
              class="col-1 text-center"
              scope="col"
            >
               # 
            </th>
            <th
              class="col-3 text-wrap-break"
              scope="col"
            >
              Title
            </th>
            <th
              class="col-4 text-wrap-break"
              scope="col"
            >
              Description
            </th>
            <th
              class="col-2 text-center"
              scope="col"
            >
              Completed
            </th>
            <th
              class="col-2 text-center"
              scope="col"
            >
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="text-center"
              colspan="5"
            >
              No todos found
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
